### PR TITLE
refactor: move MLS code from NotificationService to MLSService

### PR DESCRIPTION
**Pull request by aweiss-dev**
## Pull Request Checklist

- [ ] My code is covered by tests
- [ ] I will [merge the PR as breaking change](https://github.com/wireapp/wire-web-packages/wiki/Releases#create-major-release), if the API contract changes

### CODE REVIEW

File: packages/core/src/mls/MLSService/MLSService.ts [hunk 1]
```diff
@@ -354,11 +357,96 @@ export class MLSService {
...
-    source: PayloadBundleSource,
-    dryRun: boolean = false,
-  ): EventHandlerResult {
-    return handleBackendEvent({event, source, dryRun, mlsService: this});
+  public async handleMLSEvent(params: Omit<EventHandlerParams, 'mlsService'>): EventHandlerResult {
+    return handleBackendEvent({...params, mlsService: this});
+  }
+
+  /**
+   * ## MLS only ##
```
Comment (atomrc) [inline packages/core/src/mls/MLSService/MLSService.ts:365]: you can remove those comments 
```suggestion
```

File: packages/core/src/mls/MLSService/MLSService.ts [hunk 2]
```diff
@@ -354,11 +357,96 @@ export class MLSService {
...
+    const groupId = await this.groupIdFromConversationId?.(conversationQualifiedId);
+
+    if (!groupId) {
+      throw new Error(`Could not find a group_id for conversation ${conversationId}@${conversationDomain}`);
+    }
+    return groupId;
+  }
+
+  /**
+   * ## MLS only ##
```
Comment (atomrc) [inline packages/core/src/mls/MLSService/MLSService.ts:382]:
```suggestion
```

File: packages/core/src/mls/MLSService/MLSService.ts [hunk 3]
```diff
@@ -354,11 +357,96 @@ export class MLSService {
...
+        firingDate,
+        key: groupId,
+      });
+    } else {
+      await this.commitPendingProposals({groupId, skipDelete: true});
+    }
+  }
+
+  /**
+   * ## MLS only ##
```
Comment (atomrc) [inline packages/core/src/mls/MLSService/MLSService.ts:411]:
```suggestion
```

File: packages/core/src/mls/MLSService/MLSService.ts [hunk 4]
```diff
@@ -354,11 +357,96 @@ export class MLSService {
...
+        TaskScheduler.cancelTask(groupId);
+        pendingProposalsStore.deleteItem({groupId});
+      }
+    } catch (error) {
+      this.logger.error(`Error while committing pending proposals for groupId ${groupId}`, error);
+    }
+  }
+
+  /**
+   * ## MLS only ##
```
Comment (atomrc) [inline packages/core/src/mls/MLSService/MLSService.ts:431]:
```suggestion
```

File: packages/core/src/mls/MLSService/stores/pendingProposalsStore/pendingProposalsStore.ts [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+import {CommonMLS} from '../../../types';
+
+export type PendingProposalsParams = {
+  firingDate: number;
+} & CommonMLS;
```
Comment (atomrc) [inline packages/core/src/mls/MLSService/stores/pendingProposalsStore/pendingProposalsStore.ts:24]: Can you check if it still makes sense to have this `CommonMLS`? I feel like it's only used here (in which case we can inline the `groupId` in this type

File: packages/core/src/mls/MLSService/stores/pendingProposalsStore/pendingProposalsStore.ts [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+
+const getAllItems = (): PendingProposalsParams[] => {
+  const storedStateMap = getUpdateDatesMap();
+  return Object.values(storedStateMap);
+};
+
+const getItem = ({groupId}: CommonMLS): PendingProposalsParams | undefined => {
+  const storedState = getUpdateDatesMap();
+  return storedState[groupId];
+};
```
Comment (atomrc) [inline packages/core/src/mls/MLSService/stores/pendingProposalsStore/pendingProposalsStore.ts:48]: Seems to be only used in tests, can probably be removed if there is not real life scenario where it's used

**Review by atomrc — CHANGES_REQUESTED** [review]
Look good overall. Just a few things here and there but nice work

File: packages/core/src/mls/EventHandler/ConversationEvent/ConversationEvent.ts [hunk 7]
```diff
@@ -17,34 +17,20 @@
...
 
 const handleConversationEvent = async (params: EventHandlerParams): EventHandlerResult => {
-  const event = params.event;
-  if (isWelcomeMessage(event)) {
+  const {event} = params;
+  if (isWelcomeMessageEvent(event)) {
     return handleWelcomeMessage({...params, event});
   }
+  if (isMLSMessageAddEvent(event)) {
+    return handleMLSMessageAdd({...params, event});
```
[the remark this answers is not in the document]
Comment (aweiss-dev) [reply packages/core/src/mls/EventHandler/ConversationEvent/ConversationEvent.ts:32]: This is on purpose. Typescript doesn't recognize the right type if you try to just pass props or if props.event gets checked by the type guards.

[the remark this answers is not in the document]
Comment (atomrc) [reply packages/core/src/mls/EventHandler/ConversationEvent/ConversationEvent.ts:32]: what if we do `isMLSMessageAddEvent(params.event)`, would typescript understand that `params.event` is of the right type?

**Review by atomrc — APPROVED** [review]

**Pull request merged**